File: Node-JS-Practice/src/models/user.model.ts
export default class User {
  public ID?: string;
  public USERNAME?: string;
  public PASSWORD?: string;
  public FIRST_NAME?: string;
  public LAST_NAME?: string;
  public EMAIL?: string;
  public PHONE_NUMBER?: string;
  public STATUS?: string;
  public ROLE?: string;
  public IS_VERIFIED?: boolean;
  public IS_LOCKED?: boolean;
  public PROFILE_PIC?: string;
  public EXTRA_PROPERTIES?: string;
  public INSTITUTEID?: string;
  constructor(ID?: string, USERNAME?: string, PASSWORD?: string , FIRST_NAME?: string, LAST_NAME?: string,
    EMAIL?: string, PHONE_NUMBER?: string, STATUS?: string, ROLE?: string, IS_VERIFIED?: boolean,
    IS_LOCKED?: boolean, PROFILE_PIC?: string, EXTRA_PROPERTIES?: string,
    INSTITUTEID?: string) {

    this.ID=ID;
    this.USERNAME=USERNAME;
    this.PASSWORD=PASSWORD;
    this.FIRST_NAME=FIRST_NAME;
    this.LAST_NAME=LAST_NAME;
    this.EMAIL=EMAIL;
    this.PHONE_NUMBER=PHONE_NUMBER;
    this.STATUS=STATUS;
    this.ROLE=ROLE;
    this.IS_VERIFIED=IS_VERIFIED;
    this.IS_LOCKED= IS_LOCKED;
    this.PROFILE_PIC=PROFILE_PIC;
    this.EXTRA_PROPERTIES=EXTRA_PROPERTIES;
    this.INSTITUTEID=INSTITUTEID

  }

}

File: ZAB_FYP_PORTAL FLUTTER FRONTEND/lib/models/user_model.dart
class UserModel {
  int? id;
  String? username;
  String? firstName;
  String? lastName;
  String? email;
  String? role;
  String? phoneNumber;
  String? status;
  int? isVerified;
  int? isLocked;
  dynamic? extraProperties;
  String? instituteid;

  UserModel(
      {this.id,
      this.username,
      this.firstName,
      this.lastName,
      this.email,
      this.role,
      this.phoneNumber,
      this.status,
      this.isVerified,
      this.isLocked,
      this.extraProperties,
      this.instituteid});

  UserModel.fromJson(Map<String, dynamic> json) {
    print("json");
    print(json);
    id = json['id'];
    username = json['username'];
    firstName = json['first_name'];
    lastName = json['last_name'];
    email = json['email'];
    role = json['role'];
    phoneNumber = json['phone_number'];
    status = json['status'];
    isVerified = json['is_verified'];
    isLocked = json['is_locked'];
    extraProperties = json['extra_properties'];
    instituteid = json['instituteid'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['username'] = this.username;
    data['first_name'] = this.firstName;
    data['last_name'] = this.lastName;
    data['email'] = this.email;
    data['role'] = this.role;
    data['phone_number'] = this.phoneNumber;
    data['status'] = this.status;
    data['is_verified'] = this.isVerified;
    data['is_locked'] = this.isLocked;
    data['extra_properties'] = this.extraProperties;
    data['instituteid'] = this.instituteid;
    return data;
  }
}
